src/device-config.ts

```diff
@@ -352,23 +353,39 @@ export function bootConfigChangeRequired(
 	// Some devices require specific overlays, here we apply them
 	ensureRequiredOverlay(deviceType, targetBootConfig);
 
+	// Search for any unsupported values
+	_.each(targetBootConfig, (value, key) => {
+		if (
+			!configBackend.isSupportedConfig(key) &&
+			currentBootConfig[key] !== value
+		) {
+			const err = `Attempt to change blacklisted config value ${key}`;
+			logger.logSystemMessage(err, { error: err }, 'Apply boot config error');
+			throw new Error(err);
+		}
+	});
+
 	if (!_.isEqual(currentBootConfig, targetBootConfig)) {
-		_.each(targetBootConfig, (value, key) => {
-			// Ignore null check because we can't get here if configBackend is null
-			if (!configBackend!.isSupportedConfig(key)) {
-				if (currentBootConfig[key] !== value) {
-					const err = `Attempt to change blacklisted config value ${key}`;
-					logger.logSystemMessage(
-						err,
-						{ error: err },
-						'Apply boot config error',
-					);
-					throw new Error(err);
-				}
+		// Check if the only difference is the targetBootConfig not containing a special case
```

Comment:
I'd like to put a spotlight on this specific change. This code is a necessary evil for devices that upgrade their Supervisor version and have not had their ODMDATA configuration mode sent to the cloud which happens on provision. The logic to detect this case can be quiet the rabbit hole and I concluded the most robust way and understand way to make sure we catch this case is to do what I did rather then if (condition_1 && condition_2) because it gets messy quickly.